feed/script.js

```diff
@@ -0,0 +1,205 @@
+  if (!nextLink) {
+    return;
+  }
+  if (entries[0].isIntersecting && !isDataLoading) {
+    await populateActivityFeed({ category }, nextLink);
+  }
+});
+
+const addIntersectionObserver = () => {
+  intersectionObserver.observe(lastElementContainer);
+};
+
+const removeIntersectionObserver = () => {
+  intersectionObserver.unobserve(lastElementContainer);
+};
+
+// main entry
+renderFeed();
+
+function renderActivityItem(data) {
+  const item = document.createElement('li');
+  item.classList.add('activity-item');
+  item.innerHTML = '';
+  let content = '';
+  switch (data.type) {
+    case logType.EXTENSION_REQUESTS:
+      content = formatExtensionRequestsLog(data);
+      break;
+    case logType.TASK:
+      content = formatTasksLog(data);
+      break;
+    case logType.REQUEST_CREATED:
+    case logType.REQUEST_APPROVED:
+    case logType.REQUEST_REJECTED:
+      content = formatOOORequests(data);
+      break;
+    default:
+      content = 'Unknown activity type';
+  }
+  item.innerHTML = `
+    ${content}`;
+  return item;
+}
+
+function formatOOORequests(data) {
+  let text = '';
+  if (data.type === logType.REQUEST_CREATED)
+    text = `<p><strong>${data?.user || data?.createdBy
+      }</strong> raised an OOO request from <strong>${new Date(
+        data.from,
+      ).toLocaleString()}</strong> until <strong>${new Date(
+        data.until,
+      ).toLocaleString()}</strong></p>`;
+  if (data.status === 'APPROVED')
+    text = `${data?.user || data?.createdBy
+      } approved an extension request for task:`;
+  if (data.status === 'DENIED')
+    text = `${data?.user || data?.createdBy
+      } rejected an extension request for task:`;
+
+  return `<div>
+            <div class="title">
+              <img src="assets/leave.png" class="img_icon">
+              <p class="">${text}</p>
+           </div>
+           ${data.message ? `<strong>Reason:</strong> ${data.message}` : ''}
+           <p class="timestamp">
+          ${new Date(data.timestamp * 1000)}
+          </p>
+       </div>`;
+}
+
+function formatExtensionRequestsLog(data) {
+  let text = '';
+  if (data.status === 'PENDING')
+    text = `${data?.user || data?.createdBy
+      } raised an extension request for task:`;
+  if (data.status === 'APPROVED')
+    text = `${data?.user || data?.createdBy
+      } approved an extension request for task:`;
+  if (data.status === 'DENIED')
+    text = `${data?.user || data?.createdBy
+      } rejected an extension request for task:`;
+  return `<div>
+            <div class="title">
+              <img src="assets/extensionReq.png" class="img_icon">
+              <p class="">${text}</p>
+           </div>
+           <p>
+           Task:- 
+          <a href=${STATUS_BASE_URL}/tasks/${data?.taskId}>${data.title || 'Untitled Task'
+    }</a>
+          </p>
+          <p class="timestamp">
+          ${new Date(data.timestamp * 1000)}
+          </p>
+       </div>`;
+}
+
+function formatTasksLog(data) {
```

Comment:
1. If data.percentCompleted is present but data.status is not, the generated text might lack a closing </strong> tag for the percentage, potentially causing formatting issues.

2. The optional chaining (?.) isn't strictly necessary for accessing properties within computed properties like ${data?.user || data?.createdBy}, as those already handle potential undefined values gracefully.

Reply:
Noted.